crates/ethereum/evm/src/lib.rs

```diff
@@ -77,9 +77,14 @@ impl ConfigureEvmEnv for EthEvmConfig {
     type TxEnv = TxEnv;
     type Spec = SpecId;
 
-    fn tx_env(&self, transaction: &TransactionSigned, sender: Address) -> Self::TxEnv {
+    fn tx_env<T: Borrow<Self::Transaction>>(
+        &self,
+        transaction: impl Borrow<Recovered<T>>,
```

Comment:
I wonder if we should just operate on `Recovered<&T>` instead

because this Borrow forces `&Recovered<Tx>` which should be fine as long as the input here wraps a `tx: Recovered<Tx>`
but the additional Borrow on T should also cover use cases where the sender is stored separately
but effectively this is just `Recovered<&T>`
so this seems fine.

but I think we should also consider the consuming use case `Recovered<T>`
which should probably be a separate function?